src/library_wasmfs_fetch.js

```diff
@@ -67,11 +71,17 @@ mergeInto(LibraryManager.library, {
 
       // read/getSize fetch the data, then forward to the parent class.
       read: async (file, buffer, length, offset) => {
-        await getFile(file);
+        try {
+          await getFile(file);
+        } catch (status) {
+          return status === 404 ? -{{{ cDefine('ENOENT') }}} : -{{{ cDefine('EBADF') }}};
+        }
         return jsFileOps.read(file, buffer, length, offset);
       },
-      getSize: async(file) => {
-        await getFile(file);
+      getSize: async (file) => {
+        try {
+          await getFile(file);
+        } catch (e) {}
```

Comment:
It seems to me that any error should be handled here in `getSize` as well. But I am not sure how to properly do this.
For now if a file is not on the server we just return zero size. But probably we should also return -1 from `fstat` and set the appropriate `errno`. The problem is that we don't check for error from `lockedFile.getSize()` in `__syscall_newfstatat` and so we can't know that something here went wrong and return the appropriate error code.
Should this behavior be changed?

Reply:
Yes, we need to add proper error handling in all these APIs, I think. I believe @tlively has plans to work on that. For this PR, returning a 0 size is good enough for now.